packages/phone-number-privacy/common/src/interfaces/requests.ts

```diff
@@ -150,35 +146,156 @@ export type DomainQuotaStatusRequest<
+        const canary = (x: never) => x
+        canary(value.domain)
+        throw new Error('Implementation error: validated domain is not of any known type')
+      })
+    )
+
+  return new t.Type<DomainRestrictedSignatureRequest<D>, DomainRestrictedSignatureRequest<D>>(
+    `DomainRestrictedSignatureRequestSchema<${domain.name}>`,
+    (unk: unknown): unk is DomainRestrictedSignatureRequest<D> => isRight(validation(unk, [])),
+    validation,
+    (req: DomainRestrictedSignatureRequest<D>) => req
+  )
+}
+
+/** Parameterized schema for checking unknown input against DomainQuotaStatusRequest */
+export function domainQuotaStatusRequestSchema<D extends Domain = Domain>(
+  domain: t.Type<D>
+): t.Type<DomainQuotaStatusRequest<D>> {
+  // The schema defined here does most of the work, but does not guarantee consistency between the
+  // domain and options fields. We wrap the schema below to add a consistency check.
+  const schema = t.strict({
+    domain,
+    options: t.unknown,
+    sessionID: eip712OptionalSchema(t.string),
+  })
+
+  const validation = (unk: unknown, ctx: t.Context): t.Validation<DomainQuotaStatusRequest<D>> =>
+    pipe(
+      schema.validate(unk, ctx),
+      chain((value: t.TypeOf<typeof schema>) => {
+        if (isSequentialDelayDomain(value.domain)) {
+          const either = SequentialDelayDomainOptionsSchema.validate(value.options, ctx)
+          if (isRight(either)) {
+            return t.success(value as DomainQuotaStatusRequest<D>)
+          }
+
+          return t.failure(unk, ctx, 'options type does not match domain type')
+        }
+
+        // canary provides a compile-time check that all subtypes of Domain have branches. If a case
+        // was missed, then an error will report that domain cannot be assigned to type `never`.
+        const canary = (x: never) => x
+        canary(value.domain)
+        throw new Error('Implementation error: validated domain is not of any known type')
+      })
+    )
+
+  return new t.Type<DomainQuotaStatusRequest<D>, DomainQuotaStatusRequest<D>>(
+    `DomainQuotaStatusRequestSchema<${domain.name}>`,
+    (unk: unknown): unk is DomainQuotaStatusRequest<D> => isRight(validation(unk, [])),
+    validation,
+    (req: DomainQuotaStatusRequest<D>) => req
+  )
+}
+
+/** Parameterized schema for checking unknown input against DisableDomainRequest */
+export function disableDomainRequestSchema<D extends Domain = Domain>(
+  domain: t.Type<D>
+): t.Type<DisableDomainRequest<D>> {
+  // The schema defined here does most of the work, but does not guarantee consistency between the
+  // domain and options fields. We wrap the schema below to add a consistency check.
+  const schema = t.strict({
+    domain,
+    options: t.unknown,
+    sessionID: eip712OptionalSchema(t.string),
+  })
+
+  const validation = (unk: unknown, ctx: t.Context): t.Validation<DisableDomainRequest<D>> =>
+    pipe(
```

Comment:
I'm not sure I completely understand this use of pipe and chain. We can go over this in our 1:1